rust/src/wrapper/chart/chart_axis.rs

```diff
@@ -0,0 +1,78 @@
+use std::sync::{Arc, Mutex};
+
+use rust_xlsxwriter as xlsx;
+use wasm_bindgen::prelude::*;
+
+use super::chart_format::ChartFormat;
+
+#[derive(Copy, Clone)]
+pub enum AxisType {
+    X,
+    Y,
+    X2,
+    Y2,
+}
+
+impl ChartAxis {
+    fn with_chart<F>(&self, f: F) -> ChartAxis
+    where
+        F: FnOnce(&mut xlsx::ChartAxis),
+    {
+        let mut chart = self.inner.lock().unwrap();
+        let axis = match self.axis {
+            AxisType::X => chart.x_axis(),
+            AxisType::Y => chart.y_axis(),
+            AxisType::X2 => chart.x2_axis(),
+            AxisType::Y2 => chart.y2_axis(),
+        };
+        f(axis);
+        ChartAxis {
+            inner: Arc::clone(&self.inner),
+            axis: self.axis,
+        }
+    }
+}
+
+#[wasm_bindgen]
+pub struct ChartAxis {
+    pub(crate) inner: Arc<Mutex<xlsx::Chart>>,
+    pub(crate) axis: AxisType,
+}
+
+#[wasm_bindgen]
+impl ChartAxis {
+    #[wasm_bindgen(js_name = "setName", skip_jsdoc)]
+    pub fn set_name(&self, name: &str) -> ChartAxis {
+        self.with_chart(|axis| {
+            axis.set_name(name);
+        })
+    }
+
+    #[wasm_bindgen(js_name = "setFormat", skip_jsdoc)]
+    pub fn set_format(&self, format: &mut ChartFormat) -> ChartAxis {
+        self.with_chart(|axis| {
+            axis.set_format(&mut format.inner);
+        })
+    }
```

Comment:
Passing a mutable reference on the JavaScript side is really complicated. I think some of the similar functions in this repository are buggy. If you don't need this method at the moment, it might be better to remove it.